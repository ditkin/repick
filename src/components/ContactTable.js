import React, { Component } from 'react';
import { Table } from 'react-bootstrap';
import ContactRow from './ContactRow';

class ContactList extends Component {

    constructor(){
        super();
        this.state = {
            activeRow: null
        };
    };

    setActiveRow(contact_id) {
        this.setState({activeRow: contact_id})
    }

    _getContacts(contacts) {
        let contactTableAttributes = ['first_name', 'last_name', 'email_address', 'create_source']

        return contacts.map((contact) => {
            return (<ContactRow
                id={contact.contact_id}
                first_name={contact.first_name}
                last_name={contact.last_name}
                email_address={contact.email_address.address}
                create_source={contact.create_source || 'Added by you'}
                key={contact.contact_id}
                contactAttributes={contactTableAttributes}
                active={this.state.activeRow === contact.contact_id}
                accountId={this.props.accountId}
                setActiveRow={this.setActiveRow.bind(this)}/>);
        });
    }

    render() {
        const { contacts } = this.props;
        let contactRows = this._getContacts(contacts);

        return (
                <Table hover className="Table Table--cellBorders">
                    <thead className="Table-head">
                    <tr>
                        <th className="u-tableCellCollapse">
                            <input type="checkbox" className="Form-checkbox" id="checkbox1"></input>
                            <label htmlFor="checkbox1" className="u-paddingFlush u-block"></label>
                        </th>
                        <th>First Name</th>
                        <th>Last Name</th>
                        <th>Email Address</th>
                        <th>Source</th>
                    </tr>
                    </thead>
                    <tbody>
                    {contactRows}
                    </tbody>
                </Table>
        );
    }
}

export default ContactList;
