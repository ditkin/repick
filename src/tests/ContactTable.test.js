import React from "react";
import ContactTable from "../components/ContactTable";
import renderer from 'react-test-renderer';
import contacts_data from './sample_contacts'

it('renders list of contacts', () => {
    const div = document.createElement('div');
    const component = renderer.create(
        <ContactTable contacts={contacts_data} />,
        div
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
