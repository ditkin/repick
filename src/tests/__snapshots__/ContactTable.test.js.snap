exports[`test renders list of contacts 1`] = `
<table
  className="Table Table--cellBorders table table-hover">
  <thead
    className="Table-head">
    <tr>
      <th
        className="u-tableCellCollapse">
        <input
          className="Form-checkbox"
          id="checkbox1"
          type="checkbox" />
        <label
          className="u-paddingFlush u-block"
          htmlFor="checkbox1" />
      </th>
      <th>
        First Name
      </th>
      <th>
        Last Name
      </th>
      <th>
        Email Address
      </th>
      <th>
        Source
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <input
          className="Form-checkbox"
          data-qe-id="Table-CHECKBOX-ID"
          id="c0a921c6-fcdd-7c65-ad54-aada7138f123"
          type="checkbox" />
        <label
          className="u-paddingFlush u-block"
          htmlFor="c0a921c6-fcdd-7c65-ad54-aada7138f123" />
      </td>
      <td
        className="first_name">
        Ana
      </td>
      <td
        className="last_name">
        Rocha
      </td>
      <td
        className="email_address">
        table_1_there@nullmailer.com
      </td>
      <td
        className="create_source">
        Added by you
      </td>
    </tr>
    <tr>
      <td>
        <input
          className="Form-checkbox"
          data-qe-id="Table-CHECKBOX-ID"
          id="e605a263-54cc-4dbf-2432-5622ca2d946a"
          type="checkbox" />
        <label
          className="u-paddingFlush u-block"
          htmlFor="e605a263-54cc-4dbf-2432-5622ca2d946a" />
      </td>
      <td
        className="first_name">
        Judith
      </td>
      <td
        className="last_name">
        Reeves
      </td>
      <td
        className="email_address">
        from_2_these@nullmailer.com
      </td>
      <td
        className="create_source">
        Added by you
      </td>
    </tr>
  </tbody>
</table>
`;
